### PR TITLE
fix(cloud-sql): update samples to show TCP and Unix Socket connection methods

### PR DESCRIPTION
**Pull request by shubha-rajan**
* Updated both Postgres and MySQL samples to show connection with both TCP and Unix sockets
* Updated READMEs
* Updated tests to test whether connection is successful
* Updated build script to run proxy with both TCP and unix socket connection methods

### CODE REVIEW

File: .kokoro/build.sh [hunk 1]
```diff
@@ -99,10 +99,13 @@ if [[ $SQL_CLIENT ]]; then
 	wget --quiet https://dl.google.com/cloudsql/cloud_sql_proxy.linux.amd64 -O cloud_sql_proxy
 	chmod +x cloud_sql_proxy	
 	if [[ $SQL_CLIENT == 'sqlserver' ]]; then
-		./cloud_sql_proxy -instances="${CONNECTION_NAME}"=tcp:1433 &>> cloud_sql_proxy.log &
+		./cloud_sql_proxy -instances="${INSTANCE_CONNECTION_NAME}"=tcp:1433 &>> cloud_sql_proxy.log &
+	elif [[ $SQL_CLIENT == 'mysql' ]]; then
+		export DB_HOST=127.0.0.1:3306
+		./cloud_sql_proxy -instances="${INSTANCE_CONNECTION_NAME}"=tcp:3306 &>> cloud_sql_proxy.log &
 	else
-		mkdir /cloudsql; chmod 777 /cloudsql
```
Comment (grayside) [inline .kokoro/build.sh:104]: Did something change about needing to pre-create & permission this directory?

File: cloud-sql/postgres/knex/test/createTable.test.js [hunk 2]
```diff
@@ -22,18 +22,28 @@ const {exec} = require('child_process');
 const cwd = path.join(__dirname, '..');
 
 const {DB_USER, DB_PASS, DB_NAME} = process.env;
-const CONNECTION_NAME = process.env.CLOUD_SQL_CONNECTION_NAME;
-
+const CONNECTION_NAME = process.env.INSTANCE_CONNECTION_NAME;
+const DB_HOST = process.env.DB_HOST;
```
Comment (grayside) [inline cloud-sql/postgres/knex/test/createTable.test.js:26]: nit: Why not include this in the destructuring statement above?

File: cloud-sql/postgres/knex/test/createTable.test.js [hunk 3]
```diff
@@ -22,18 +22,28 @@ const {exec} = require('child_process');
...
   try {
     const knex = Knex({
       client: 'pg',
-      connection: {
-        user: DB_USER,
-        password: DB_PASS,
-        database: DB_NAME,
-        host: `/cloudsql/${CONNECTION_NAME}`,
-      },
+      connection: connection,
```
Comment (grayside) [inline cloud-sql/postgres/knex/test/createTable.test.js:46]: I don't recall the practice in this repo, should this use object property shorthand?

File: cloud-sql/postgres/knex/test/createTable.test.js [hunk 4]
```diff
@@ -43,7 +53,7 @@ before(async () => {
 
 it('should create a table', (done) => {
   exec(
-    `node createTable.js ${DB_USER} ${DB_PASS} ${DB_NAME} ${CONNECTION_NAME}`,
+    `node createTable.js ${DB_USER} ${DB_PASS} ${DB_NAME} ${CONNECTION_NAME} ${DB_HOST}`,
```
Comment (grayside) [inline cloud-sql/postgres/knex/test/createTable.test.js:56]: Is there a plan to eventually move away from the CLI mechanism?

File: .kokoro/build.sh (hunk 1 above)
Comment (shubha-rajan) [reply .kokoro/build.sh:104]: I changed it to run the tests using TCP instead, so there's no need for a socket directory. I noticed that there were issues with writing to the root directory when the tests started failing once I changed them to run against an actual db connection

File: cloud-sql/postgres/knex/test/createTable.test.js (hunk 2 above)
Comment (shubha-rajan) [reply cloud-sql/postgres/knex/test/createTable.test.js]: updated in latest commit

File: cloud-sql/postgres/knex/test/createTable.test.js (hunk 3 above)
Comment (shubha-rajan) [reply cloud-sql/postgres/knex/test/createTable.test.js]: I updated it in the latest commit, but I'm not sure what the repo standard is.

File: cloud-sql/postgres/knex/test/createTable.test.js (hunk 4 above)
Comment (shubha-rajan) [reply cloud-sql/postgres/knex/test/createTable.test.js:55]: The MySQL and SQL Server samples have an `ensureSchema` method that creates a table if it doesn't exist each time `server.js` is run. Should we switch to using that approach here?

File: .kokoro/build.sh (hunk 1 above)
Comment (grayside) [reply .kokoro/build.sh:104]: Got it, so we're not doing end-to-end tests where the code is deployed to GAE or Cloud Run?

File: cloud-sql/postgres/knex/test/createTable.test.js (hunk 4 above)
Comment (grayside) [reply cloud-sql/postgres/knex/test/createTable.test.js:55]: I think that would be a good idea but it seems lower priority. Probably something for you/your team to decide.

**Comment by grayside** [comment]
(Note I approved but did not test that the code continues to work)

File: .kokoro/build.sh (hunk 1 above)
Comment (shubha-rajan) [reply .kokoro/build.sh:104]: Not yet, anyway. Setting up GAE and Cloud Run e2e tests for all Cloud SQL samples is on the roadmap.

File: cloud-sql/postgres/knex/test/createTable.test.js (hunk 4 above)
Comment (kurtisvg) [reply cloud-sql/postgres/knex/test/createTable.test.js:55]: +1 - probably best in a separate PR.

File: cloud-sql/postgres/knex/server.js [hunk 5]
```diff
@@ -44,65 +44,93 @@ const logger = winston.createLogger({
...
-  // something like https://cloud.google.com/kms/ to help keep secrets secret.
-  const config = {
-    user: process.env.DB_USER, // e.g. 'my-user'
-    password: process.env.DB_PASS, // e.g. 'my-user-password'
-    database: process.env.DB_NAME, // e.g. 'my-database'
-  };
+// [START cloud_sql_postgres_knex_create_tcp]
+const connectWithTcp = (config) => {
+  // Extract host and port from socket address
+  const dbSocketAddr = process.env.DB_HOST.split(":")
```
Comment (kurtisvg) [inline cloud-sql/postgres/knex/server.js:50]:
```suggestion
  const dbSocketAddr = process.env.DB_HOST.split(":") // e.g. '127.0.0.1:5432'
```

**Comment by shubha-rajan** [comment]
I just submitted a CL to run the tests for postgres and it looks like they're failing. I'll investigate and fix that before merging.

**Comment by shubha-rajan** [comment]
Tests are passing now! Had to go change the instance name for the database.

**Pull request merged**